File: sdk/typescript/src/finops.ts
/**
 * TypeScript SDK Client for FinOps Intelligence Platform (Phase 5.42).
 */

export class FinOpsClient {
  private baseUrl: string;

  constructor(baseUrl: string = "http://localhost:8000") {
    this.baseUrl = baseUrl.replace(/\/$/, "");
  }

  async listCosts(tenantId: string = "global"): Promise<any[]> {
    const res = await fetch(`${this.baseUrl}/v1/finops/costs?tenant_id=${tenantId}`);
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }

  async recordCost(tenantId: string, category: string, amountUsd: number): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/finops/costs`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ tenant_id: tenantId, category, amount_usd: amountUsd }),
    });
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }

  async listBudgets(tenantId: string = "global"): Promise<any[]> {
    const res = await fetch(`${this.baseUrl}/v1/finops/budgets?tenant_id=${tenantId}`);
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }

  async createBudget(tenantId: string, name: string, amountUsd: number): Promise<Record<string, any>> {
    const res = await fetch(`${this.baseUrl}/v1/finops/budgets`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ tenant_id: tenantId, name, amount_usd: amountUsd }),
    });
    if (!res.ok) throw new Error(`Fetch failed: ${res.statusText}`);
    return res.json();
  }
}

export const FinOpsIntelligenceClient = FinOpsClient;

File: sdk/go/finops.go
package llmengine

import (
	"encoding/json"
	"fmt"
	"net/http"
)

type FinOpsClient struct {
	BaseURL    string
	HTTPClient *http.Client
}

func NewFinOpsClient(baseURL string) *FinOpsClient {
	return &FinOpsClient{
		BaseURL:    baseURL,
		HTTPClient: &http.Client{},
	}
}

func (c *FinOpsClient) GetCosts(tenantID string) ([]map[string]interface{}, error) {
	resp, err := c.HTTPClient.Get(fmt.Sprintf("%s/v1/finops/costs?tenant_id=%s", c.BaseURL, tenantID))
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var result []map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, err
	}
	return result, nil
}
